src/Parquet.chpl

```diff
@@ -48,62 +76,98 @@ module Parquet {
+  proc getArrSize(filename: string) throws {
+    extern proc c_getNumRows(chpl_str, errMsg): int;
+    var pqErr = new parquetErrorMsg();
+    
+    var size = c_getNumRows(filename.localize().c_str(),
+                            c_ptrTo(pqErr.errMsg));
+    if size == ARROWERROR {
+      pqErr.parquetError(getLineNumber(), getRoutineName(), getModuleName());
+    }
     return size;
   }
 
-  proc getArrType(filename: string, colname: string) {
-    extern proc c_getType(filename, colname): c_int;
+  proc getArrType(filename: string, colname: string) throws {
+    extern proc c_getType(filename, colname, errMsg): c_int;
+    var pqErr = new parquetErrorMsg();
     var arrType = c_getType(filename.localize().c_str(),
-                            colname.localize().c_str());
+                            colname.localize().c_str(),
+                            c_ptrTo(pqErr.errMsg));
+    if arrType == ARROWERROR {
+      pqErr.parquetError(getLineNumber(), getRoutineName(), getModuleName());
+    }
+    
     if arrType == ARROWINT64 then return ArrowTypes.int64;
     else if arrType == ARROWINT32 then return ArrowTypes.int32;
     return ArrowTypes.notimplemented;
   }
 
   proc writeDistArrayToParquet(A, filename, dsetname, rowGroupSize) throws {
     extern proc c_writeColumnToParquet(filename, chpl_arr, colnum,
-                                     dsetname, numelems, rowGroupSize);
+                                       dsetname, numelems, rowGroupSize,
+                                       errMsg): int;
     var filenames: [0..#A.targetLocales().size] string;
     for i in 0..#A.targetLocales().size {
       var suffix = '%04i'.format(i): string;
       filenames[i] = filename + "_LOCALE" + suffix + ".parquet";
     }
+    var matchingFilenames = glob("%s_LOCALE*%s".format(filename, ".parquet"));
 
+    var warnFlag = processParquetFilenames(filenames, matchingFilenames);
+    
     coforall (loc, idx) in zip(A.targetLocales(), filenames.domain) do on loc {
+        var pqErr = new parquetErrorMsg();
         const myFilename = filenames[idx];
 
         var locDom = A.localSubdomain();
         var locArr = A[locDom];
-        c_writeColumnToParquet(myFilename.localize().c_str(), c_ptrTo(locArr), 0, dsetname.localize().c_str(), locDom.size, rowGroupSize);
+        if c_writeColumnToParquet(myFilename.localize().c_str(), c_ptrTo(locArr), 0,
+                                  dsetname.localize().c_str(), locDom.size, rowGroupSize,
+                                  c_ptrTo(pqErr.errMsg)) == ARROWERROR {
+          pqErr.parquetError(getLineNumber(), getRoutineName(), getModuleName());
+        }
       }
+    return warnFlag;
+  }
+      
+
+  proc processParquetFilenames(filenames: [] string, matchingFilenames: [] string) throws {
+    var warnFlag: bool;
+    if matchingFilenames.size > 0 {
+      warnFlag = true;
+    } else {
+      warnFlag = false;
+    }
+    return warnFlag;
```

Comment:
In theory you could condense this to `return matchingFilenames.size > 0;`

Also, this seems like a weird bit of logic for a proc named `processParquetFilenames`, is something supposed to be done to the filenames?  Is this a placeholder for something later?

Reply:
Ya, this is based on the `processFilenames` function in `GenSymIO` and the idea is to handle the append case in there eventually, but since it isn't supported today it is a pretty minimal function just checking if we are overwriting any existing files.

This reminds me of something that I have been thinking of around supporting append for Parquet files though -- so, Parquet does not have an explicit function to just append something to a file, so we would need to read in the entire file and then append the column onto the read file and append the data like that before writing it back out to the file. With that in mind, is supporting append for Parquet still something that would be valuable, given the performance wouldn't be comparable with the regular case? (this may not be the best place to have this discussion)

Reply:
I'm not sure about the append case, but for the current `processParquetFilenames` impl, I'm good with your explanation.